File: 6 Zigzag Conversion/6zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s
        c = 1
        sign = -1
        news = {}

        for _ in range(numRows):
            news[c] = ""
            c += 1

        c = 1

        for i in s:
            news[c] += i
            
            if c == numRows or c == 1:
                sign *= -1

            c += sign

        res = ""
        for i in news:
            res += news[i]

        return res

File: 6 Zigzag Conversion/6zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1){return s;}
        string a, b, c, res;
        map<int, string> dict;

        for (int idx = 0; idx < numRows; idx++){dict[idx] = "";}

        int i = 0, m = 1;

        for (int idx = 0; idx < s.size(); idx++){
            if (i < 0){
                m = 1;
                i = 1;
            } else if (i == numRows){
                m = -1;
                i = numRows - 2;
            }
            dict[i] += s[idx];
            i += m;
        }

        for (int idx = 0; idx < numRows; idx++){
            res += dict[idx];
        }
        return res;

    }
};